### PR TITLE
[download][s]Support for Column Order by header parameter 

### PR DESCRIPTION
**Pull request by leomrocha**
* Fixes `Error 500` when making an invalid format download request. 
* Adds more details to the existing download tests
* Now supports sorting the returned columns by a given order if in the `/download` query:

Query must be a **POST** and have the following fields:
```json
{
body: MY_GRAPHQL_QUERY,
header: [col1,col2,col3]
}
```
and contain a valid list of column names including all the `columns (fields)` in the graphql query

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -121,12 +121,11 @@ DOWNLOAD_FORMATS_SUPPORTED = ['json', 'csv', 'tsv', 'xlsx', 'ods']
 router.post(`/${APP_VERSION}/download`, async function (req, res, next) {
   // get the graphql query from body
   const query = req.body.query ? req.body.query : req.body
+  const xlsxHeader = req.body.header ? req.body.header : undefined
+  // TODO header verification, else it will break when ordering fields
```
Comment (EvgeniiaVak) [inline routes/index.js:125]: @leomrocha still a todo?

File: routes/index.js [hunk 2]
```diff
@@ -153,7 +158,9 @@ router.post(`/${APP_VERSION}/download`, async function (req, res, next) {
 
       //iterate over the result sets and create a work sheet to append to the book
       Object.keys(gqlRes).map((k) => {
-        const ws = XLSX.utils.json_to_sheet(gqlRes[k])
+        const ws = xlsxHeader
+          ? XLSX.utils.json_to_sheet(gqlRes[k], { header: xlsxHeader })
```
Comment (EvgeniiaVak) [inline routes/index.js:162]: `json_to_sheet` is pythonic style, JS uses camelCase usually

File: routes/index.js [hunk 3]
```diff
@@ -153,7 +158,9 @@ router.post(`/${APP_VERSION}/download`, async function (req, res, next) {
...
+          : XLSX.utils.json_to_sheet(gqlRes[k])
         XLSX.utils.book_append_sheet(wb, ws, k)
```
Comment (EvgeniiaVak) [inline routes/index.js:164]: same here

File: routes/index.js (hunk 2 above)
Comment (leomrocha) [reply routes/index.js:162]: json_to_sheet is the external library, we can't change it

File: routes/index.js (hunk 3 above)
Comment (leomrocha) [reply routes/index.js:164]: json_to_sheet is the external library, we can't change it

File: routes/index.js (hunk 1 above)
Comment (leomrocha) [reply routes/index.js:125]: Yes, we should do that when we are able to also validate the GraphQL query too.

**Pull request merged**